germanium/crawler.py

```diff
@@ -89,32 +98,34 @@ def run(self):
             self._call_request(self.urls.pop())
 
     def _parse_urls(self, url, resp):
-        content = resp.content
-        if 'utf-8' in resp['Content-Type']:
-            content = content.decode('utf-8')
-
         returned_urls = []
-
-        for link in self.link_extractors.get(resp['Content-Type'], self.link_extractors['default']).extract(content):
-            parsed_href = urlparse(link)
-
-            if not parsed_href.path:
-                continue
-
-            if parsed_href.scheme and not parsed_href.netloc.startswith('testserver'):
-                LOG.debug('Skipping external link: %s', link)
-                continue
-
-            if ('django.contrib.staticfiles' in settings.INSTALLED_APPS
-                    and parsed_href.path.startswith(settings.STATIC_URL)):
-                LOG.debug('Skipping static file %s', parsed_href.path)
-            elif parsed_href.path.startswith(settings.MEDIA_URL):
-                LOG.debug('Skipping media file %s', parsed_href.path)
-            elif parsed_href.path.startswith('/'):
-                returned_urls.append(link)
-            else:
-                # We'll use urlparse's urljoin since that handles things like <a href="../foo">
-                returned_urls.append(urljoin(url, link))
+        content_type = resp['Content-Type'].split(';')[0]
+        if content_type in self.link_extractors:
+            content = resp.content
+            if 'utf-8' in resp['Content-Type']:
+                content = content.decode('utf-8')
+
+            for link in self.link_extractors.get(resp['Content-Type'].split(';')[0],
+                                                 self.link_extractors['default']).extract(content):
```

Comment:
The default value should be just `'default'` not `self.link_extractors['default']`.